### PR TITLE
Redis structure change

### PR DESCRIPTION
**Pull request by nimeshkverma**
@takirohit 

Made the changes according to the new redis structure added ttl functionality. The code is working fine and is refactored. 
**Please review the code.
ASAP.**

Thanks

### CODE REVIEW

File: cartman.py [hunk 1]
```diff
@@ -18,17 +18,25 @@ def __init__(self, user_id):
             Constructor for the class, initializes user_id and checks whether
             the users' cart exists or not.
         """
+        self.redis_user_hash_token = "CARTMAN"
```
Comment (takirohit) [inline cartman.py]: Its should be constant. Do not pollute constructor

Comment (rohitkhatana) [inline cartman.py]: And should be private class variable

File: cartman.py [hunk 2]
```diff
@@ -74,21 +82,24 @@ def add(self, product_id, unit_cost, quantity=1, **extra_data_dict):
...
-        product_dict = self.self.__product_dict(
+        product_dict = self.__product_dict(
             unit_cost, quantity, extra_data_dict)
         is_added = self.redis_connection.hset(
             self.user_redis_key, product_id, Serializer.dumps(product_dict))
-        if not self.user_cart_exists and is_added:
-            self.user_cart_exists = True
-        return True if is_added else False
+        self.user_cart_exists = self.cart_exists(self.user_id)
+        if is_added or self.user_cart_exists:
```
Comment (rohitkhatana) [inline cartman.py]: instead of boolean check throw error and why are you checking for cart again??

**Comment by nimeshkverma** [comment]
@takirohit 

Updated the code as per as the review, please test it on your local and pass :)

**Comment by nimeshkverma** [comment]
Merge the code after the code review

**Pull request merged**